makefiles/arch/riscv.inc.mk

```diff
@@ -17,12 +17,15 @@ endif
 TARGET_ARCH ?= $(TARGET_ARCH_RISCV)
 
 # define build specific options
-CFLAGS_CPU   = -march=rv32imac -mabi=ilp32 -mcmodel=medlow -msmall-data-limit=8
-CFLAGS_LINK  = -nostartfiles -ffunction-sections -fdata-sections
+CFLAGS_CPU   = -march=rv32imac -mabi=ilp32
+ifneq ($(TOOLCHAIN),llvm)
+  CFLAGS_CPU += -mcmodel=medlow -msmall-data-limit=8
+endif
```

Comment:
```suggestion
ifeq ($(TOOLCHAIN),llvm)
  # Always use riscv32-none-elf as target triple for clang, as some
  # autodetected gcc target triples are incompatible with clang
  TARGET_ARCH := riscv32-none-elf 
else
  CFLAGS_CPU += -mcmodel=medlow -msmall-data-limit=8
endif
```

Reply:
Sorry it took me a while to reply, but now I finally got around to revising this issue. Regarding your proposed change: This does fix the issue, however, in my opinion it is not a proper fix for the issue. The underlying issue here is that `riscv-none-elf` is simply not a valid tripled as `riscv` isn't an architecture (there is `riscv32` and `riscv64` but no `riscv`). From the triplet you are using clang simply can't interfere whether it is compiling for a 32 Bit or 64 Bit architecture. The proper way to fix this issue is stop using `riscv-none-elf` and use `riscv32-none-elf` (or similar) instead.

Compare:

```
$ clang -target riscv32-none-elf -c -o test.o test.c
$ echo $?
0
$ clang -target riscv-none-elf -c -o test.o test.c
error: unknown target triple 'riscv-none-unknown-elf', please use -triple or -arch
```

Reply:
Most GCC toolchains can compile for both `riscv32-none-elf` and `riscv64-none-elf`. Hence, `riscv-none-elf` is the most technically correct target triplet for a multilib GCC - or at least it is equally wrong as `riscv32-none-elf` but less confusing.

But in the end it doesn't matter what is most technically correct from a theoretical point of view. In practice, nobody is using `riscv32-none-elf-gcc`. So the current approach to just pick whatever target triple GCC is using will remain, as this is the only way this actually works in practice.

Reply:
> But in the end it doesn't matter what is most technically correct from a theoretical point of view.

Ok, if you don't care what's technically correct than let's just do what you are proposing.